demo/fake-api.js

```diff
@@ -0,0 +1,614 @@
+          },
+          StackdepthSetting: {
+            score: 25.0,
+            name: "Stackdepth Setting",
+            topic: "jvm_information",
+            description:
+              "Summary:\nSome stack traces were truncated in this recording.\n\nExplanation:\nThe Flight Recorder is configured with a maximum captured stack depth of 64. 3.11 % of all traces were larger than this option, and were therefore truncated. If more detailed traces are required, increase the \u0027\u0027-XX:FlightRecorderOptions\u003dstackdepth\u003d\u003cvalue\u003e\u0027\u0027 value.\nEvents of the following types have truncated stack traces: org.openjdk.jmc.flightrecorder.rules.jdk.general.StackDepthSettingRule$StackDepthTruncationData@21e159e2,org.openjdk.jmc.flightrecorder.rules.jdk.general.StackDepthSettingRule$StackDepthTruncationData@174930bc,org.openjdk.jmc.flightrecorder.rules.jdk.general.StackDepthSettingRule$StackDepthTruncationData@4f5d6223",
+          },
+          PasswordsInArguments: {
+            score: 0.0,
+            name: "Passwords in Java Arguments",
+            topic: "jvm_information",
+            description:
+              "Summary:\nThe recording does not seem to contain passwords in the application arguments.",
+          },
+        }
+      );
+    });
+    this.get("/api/v1/targets/:targetId/recordingOptions", () => []);
+    this.get("/api/v1/targets/:targetId/events", () => [
+      {
+        category: ["GC", "Java Virtual Machine"],
+        name: "GC Heap Configuration",
+        typeId: "jdk.GCHeapConfiguration",
+        description: "The configuration of the garbage collected heap",
+      },
+    ]);
+    this.get("/api/v1/targets/:targetId/templates", () => [
+      {
+        name: "Demo Template",
+        provider: "Demo",
+        type: "TARGET",
+        description: "This is not a real event template, but it is here!",
+      },
+    ]);
+    this.get("/api/v2/probes", () => []);
+
+    this.post("/api/v2/rules", (schema, request) => {
+      let attrs = JSON.parse(request.requestBody);
+      return {
+        data: {
+          result: schema.rules.create(attrs),
+        },
+      };
+    });
+    this.get("/api/v2/rules", (schema) => ({
+      data: { result: schema.rules.all().models },
+    }));
+    this.get("/api/v2.2/credentials", () => ({ data: { result: [] } }));
+
+    this.post("/api/v2.2/graphql", (schema, request) => {
+      let body = JSON.parse(request.requestBody);
+      let query = body.query.trim();
+      let variables = body.variables;
+      let begin = query.substring(0, query.indexOf("{"));
+      let name = "unknown";
+      for (let n of begin.split(" ")) {
+        if (n == "{") {
+          break;
+        }
+        if (!n || n == "query") {
+          continue;
+        }
+        name = n.substring(0, n.indexOf("("));
```

Comment:
I think the above is caused by this line.

There could be a case as:

```graphql
query AllTargetsArchives {
               targetNodes {
                 target {
                   serviceUri
                   alias
                 }
                 recordings {
                   archived {
                     aggregate {
                       count
                     }
                   }
                 }
               }
             }
```

Reply:
The view doesn't work because the query "parsing" is broken, but also the stubbed out data model is completely fake and just has a flat list of Targets, (Active) Recordings, and Archives. I didn't bother making this into a proper mock that actually tracks the relationship between active recordings and the targets they came from, or anything like that, because I'm not sure it's really worthwhile trying to make this little toy demo thing so full-featured and "lifelike". I can fix this query but then the view is just going to list some nonsensical data showing every fake target, each of which owning every fake archived recording, unless more effort goes into making the fake data model actually make sense.